# Microarray analysis by limma and affy R package

## Microarray wikipedia

A DNA microarray (also commonly known as DNA chip or biochip) is a collection of microscopic DNA spots attached to a solid surface. Scientists use DNA microarrays to measure the expression levels of large numbers of genes simultaneously or to genotype multiple regions of a genome. Each DNA spot contains picomoles (10âˆ’12 moles) of a specific DNA sequence, known as probes (or reporters or oligos). These can be a short section of a gene or other DNA element that are used to hybridize a cDNA or cRNA (also called anti-sense RNA) sample (called target) under high-stringency conditions. Probe-target hybridization is usually detected and quantified by detection of fluorophore-, silver-, or chemiluminescence-labeled targets to determine relative abundance of nucleic acid sequences in the target. _[Link 1][1]_

## Microarray company

+ [Illumina][2]
+ [Agilent][3]
+ [Affymetrix][4]

## Analysis pipline

+ Importing and accessing probe-level data; 
+ Exploratory Analysis/Quality Assessment;
+ Preprocessing
    + Background adjustment
    + Normalization
    + Summarization
+ Quality assessment; 
+ Differential expression; 
+ Functional analysis of large gene lists; 

## R package

+ [limma][5]
+ [affy][7]
+ [affyPLM][8]
+ [affydata][9]
+ [tkWidgets][11]

## Analysis .CEL(probe level data set) file by "affy"

### read in .CEL file

Create a directory, move all the relevant CEL files to that directory

If using linux/unix, start R in that directory.

If using the Rgui for Microsoft Windows make sure your working directory contains
the Cel files (use \File -> Change Dir" menu item).

Load the library.
    
    R> library(affy) ##load the affy package

Read in the data and create an expression, using RMA for example.

    R> Data=ReadAffy() ##read data in working directory

### Normalization

load sample data: Dilution
 
    R> source("http://bioconductor.org/biocLite.R")
    R> biocLite("affydata")
    R> library(affydata)
    R> data(Dilution)

Normalization

    R> eset <- rma(Dilution) 
    #This command use rma normalized method. or you can use:
    R> eset <- mas5(Data)

Write out normalization gene expression result

    R> write.exprs(eset, file="mydata.txt")

### Exploratory Analysis/Quality Assessment

check data structure

    R> attributes(eset)

Spatial Image

Histograms / Density plot
 
    R> png("histgrams.png")
    R> hist(eset[,1:2])
    R> dev.off()

MAplot

    R> png("maplot.png")
    R> MAplot(eset[,c(1:2)], pairs=TRUE, plot.method="smoothScatter")
    R> dev.off()

### Differential expressed gene detected by other software

\# cont.

### Function annotation by other software

\# cont.

## Analysis .gpr file by "limma" 

### load library and read in .gpr file

R command:

    R> library(limma)
    R> targets <- readTargets("targets.txt")

"targets.txt" should prepare like this:

    FileName
    GSM798892.gpr
    GSM798893.gpr
    GSM798894.gpr
    GSM798895.gpr
    GSM798896.gpr
    GSM798897.gpr

### Normalization

Set up a filter so that any spot with a flag of -99 or less gets zero weight.

    > f <- function(x) as.numeric(x$Flags > -99)

Normalize

    # read in data 
    RG <- read.maimages(targets, source="genepix", wt.fun=f, columns=list(Rf='F635 Median',Gf='F635 Median', Rb = 'B635 Median', Gb='B635 Median'))
    # background correction
    RGb <- backgroundCorrect(RG, method="subtract")
    # quantile normalization
    MA.q <- normalizeBetweenArrays(RGb, method="quantile")
QC

    # MA plot to check the quality
    plotMA(RG)


\# cont.

Ref:

[http://www.sbforum.org/cmsimages/presentations/BioC.pdf][10]
[http://shiulab.plantbiology.msu.edu/wiki/index.php/Affymetrix_processing][12]

[1]: http://en.wikipedia.org/wiki/DNA_microarray
[2]: http://www.illumina.com/applications/genotyping.ilmn
[3]: http://www.agilent.com
[4]: http://www.affymetrix.com
[5]: http://www.bioconductor.org/packages/2.12/bioc/html/limma.html
[6]: http://www.bioconductor.org/
[7]: http://www.bioconductor.org/packages/2.11/bioc/html/affy.html
[8]: http://www.bioconductor.org/packages/2.11/bioc/html/affyPLM.html
[9]: http://www.bioconductor.org/packages/2.6/data/experiment/html/affydata.html
[10]: http://www.sbforum.org/cmsimages/presentations/BioC.pdf
[11]: http://www.bioconductor.org/packages/2.10/bioc/html/tkWidgets.html
[12]: http://shiulab.plantbiology.msu.edu/wiki/index.php/Affymetrix_processing
