# Some examples about re-engineering gene regulatory network based on expression data

## Some related reviews

1. [Dialogue for Reverse Engineering Assessments and Methods][7]
2. [Wisdom of crowds for robust gene network inference][8]
3. [Gene regulatory network, wikipedia][9]

## Software for gene interaction inferring or re-engineering

+ [ARACNE][10]
+ [catnet][2]
+ [GENIE3][11]
+ others

## Examples

+ ARACNE

    If you have many gene expression microarray data or RNA-seq data, this method can work well.

    Download software:

        $ wget http://wiki.c2b2.columbia.edu/califanolab/download/ARACNE2/ARACNE.src.tar.gz .
        $ tar xzvf ARACNE.src.tar.gz
        $ cd ./ARACNE
    
    Prepare data: 

        Probe_ID        Gene_Symbol     GSM425337.CEL   GSM425338.CEL
        1415670_at      Copg            9.5180          9.6583
        1415671_at      Atp6v0d1        8.9955          9.1125
        1415672_at      Golga7          9.5823          9.6111

    Inferring:

        $ ./aracne2 -i c2c12.expr.data -p 1e-7 -e 0.15 1>std.log 2> err.log &

+ GENIE3

    If you have many gene expression microarray data or RNA-seq data, this method can work well.

    Download software:

        $ wget http://www.montefiore.ulg.ac.be/~huynh-thu/misc/GENIE3_R.zip
        $ unzip GENIE3_R.zip
        $ cd ./GENIE3_R
        $ R

    Install [randomForest][12] R [package][13]:

        R> install.package("randomForest")
        R> library(randomForest)
        R> source("genie3.R")
    
    Prepare data:

        "Itm2a" 4.27536 4.21628 4.42731
        "Sergef"        5.78297 5.77658 5.35783
        "Fam109a"       6.0614  5.8688  6.25398
        "Dhx9"  5.76787 5.63261 5.72688
        "Ssu72" 3.66481 3.52972 3.6411

    Read in expression data:

        R> expr.matrix <- read.expr.matrix("c2c12.expr.data.txt.MGI.filter.genie3.tsv", form="rows.are.genes")
    
    Inferring interaction matrix:

        R> weight.matrix <- get.weight.matrix(expr.matrix)
    
    Get top XXXXX interaction:

        R> link.list <- get.link.list(weight.matrix, report.max=100000)
    
    Write out the result:

        R> write.table(link.list,"./c2c12.interaction.top100000.tab", quote = FALSE, sep = "\t")

+ catnet

    Cont.

+ correlation method

    Cont.
    
Ref:

[http://www.graphviz.org/Documentation.php][1]

[http://rwiki.sciviews.org/doku.php?id=packages:cran:catnet][2]

[http://dream.broadinstitute.org/][4]

[http://genepattern.broadinstitute.org/gp/pages/index.jsf][6]

[http://cran.r-project.org/web/packages/catnet/index.html][3]

Last updated: Jan. 9, 2013

[1]: http://www.graphviz.org/Documentation.php
[2]: http://rwiki.sciviews.org/doku.php?id=packages:cran:catnet
[3]: http://cran.r-project.org/web/packages/catnet/index.html
[4]: http://dream.broadinstitute.org/
[5]: http://www.nature.com/nmeth/journal/v9/n8/full/nmeth.2016.html
[6]: http://genepattern.broadinstitute.org/gp/pages/index.jsf
[7]: http://www.the-dream-project.org/
[8]: https://www.dropbox.com/s/7h9a5rh7ea5cggr/Integrating_multiple_evidence_sources_to_predict_transcription_factor_binding_in_the_human_genome_Genome-Res.-2010-Ernst-526-36.pdf
[9]: http://en.wikipedia.org/wiki/Gene_regulatory_network
[10]: http://wiki.c2b2.columbia.edu/califanolab/index.php/Software/ARACNE
[11]: http://www.montefiore.ulg.ac.be/~huynh-thu/software.html
[12]: http://en.wikipedia.org/wiki/Random_forest
[13]: http://cran.r-project.org/web/packages/randomForest/index.html