# Inferring gene expression network based on RNA-seq or Micro-array data

## Synopsis

[Gene regulatory network, wikipedia][18]

## Data source

+ Single gene expression data;
+ Time series gene expression data;
+ Perturbation(gene knockout, chemical perturbation or virus infection) gene expression data;
+ Drug effect expression data;

## Network reconstruction method

+ Pearson's linear correlation coefficient & Hierarchical clustering;
+ Bayesian conditional probability tables;
+ Mutual information(Information-theoretic approaches);
+ Ordinary differential equations.

### Transcriptional network re-construction software

+ Base on clustering
    + Cluster 3.0  *[1] [1]*, *[3][3]*;
    + STEM *[2] [2]* (integrate HMM algorithm to infer co-express);

+ Bayesian Network
    + Banjo *[11][11]*;
    + Bayes Net Toolbox *[12][12]*, *[13][13]*;
    + 
+ Mutual information
    + ARACNE *[10][10]*;

+ Ordinary differential equations
    + NIR *[5] [5]*;
    + TSNI *[6] [6]*, *[9][9]*;

+ Integrated method
    + DREM 2.0 *[4][4]*;
    + Inferelator *[7] [7]*, *[8][8]*;

+ Others
    + Killer *[14][14]*;
    + ANOVA *[15][15]*;

## Reviews about Network reconstruction

1. [Dialogue for Reverse Engineering Assessments and Methods][16]
2. [Wisdom of crowds for robust gene network inference][17]
3. [Gene regulatory network, wikipedia][18]

## Some examples

[Some examples about re-engineering gene regulatory network based on expression data][24]

Ref:

[http://www.graphviz.org/Documentation.php][19]

[http://rwiki.sciviews.org/doku.php?id=packages:cran:catnet][20]

[http://dream.broadinstitute.org/][21]

[http://genepattern.broadinstitute.org/gp/pages/index.jsf][22]

[http://cran.r-project.org/web/packages/catnet/index.html][23]

Last updated: Jan. 9, 2013

+ http://bioinformatics.oxfordjournals.org/content/28/10/1376.full
+ http://bib.oxfordjournals.org/content/4/3/228.abstract
+ http://bioinformatics.oxfordjournals.org/content/22/20/2523.short
+ http://bib.oxfordjournals.org/content/8/2/109.full
+ http://www.nature.com/msb/journal/v3/n1/full/msb4100120.html
+ http://rsfs.royalsocietypublishing.org/content/1/6/857.full.pdf
+ http://www-users.cselabs.umn.edu/classes/Spring-2011/csci8980-mbds/slides/Varun_CSci8980_presentation_casestudy_grn_copyfinal.pdf
+ http://www.mybiosoftware.com/
+ http://www.nature.com/nmeth/journal/v9/n8/full/nmeth.2016.html

[1]: http://bonsai.hgc.jp/~mdehoon/software/cluster/software.htm#ctv 
[2]: http://www.cs.cmu.edu/~jernst/stem/
[3]: http://bioinformatics.oxfordjournals.org/content/21/suppl_1/i159.full.pdf+html
[4]: http://www.sb.cs.cmu.edu/drem/
[5]: http://dibernardo.tigem.it/wiki/index.php/Network_Inference_by_Reverse-engineering_NIR
[6]: http://dibernardo.tigem.it/wiki/index.php/Time_Series_Network_Identification_TSNI
[7]: http://err.bio.nyu.edu/inferelator/
[8]: http://genomebiology.com/2006/7/5/R36
[9]: http://www.seas.ucla.edu/~liaoj/download.html
[10]: http://wiki.c2b2.columbia.edu/califanolab/index.php/Software/ARACNE
[11]: http://www.cs.duke.edu/~amink/software/banjo/
[12]: http://compbio.cs.huji.ac.il/LibB/programs.html#GenInstance
[13]: https://code.google.com/p/bnt/
[14]: http://cogito-b.ml.cmu.edu/keller/ 
[15]: http://www2.bio.ifi.lmu.de/~kueffner/anova.tar.gz
[16]: http://www.the-dream-project.org/
[17]: https://www.dropbox.com/s/7h9a5rh7ea5cggr/Integrating_multiple_evidence_sources_to_predict_transcription_factor_binding_in_the_human_genome_Genome-Res.-2010-Ernst-526-36.pdf
[18]: http://en.wikipedia.org/wiki/Gene_regulatory_network
[19]: http://www.graphviz.org/Documentation.php
[20]: http://rwiki.sciviews.org/doku.php?id=packages:cran:catnet
[21]: http://dream.broadinstitute.org/
[22]: http://genepattern.broadinstitute.org/gp/pages/index.jsf
[23]: http://cran.r-project.org/web/packages/catnet/index.html 
